Add Landing Page testing recommendation

gonzalo-bulnes opened the pull request:
## Status
<!-- What state is your PR in? Select one of the following and delete the option that does not apply. -->

Ready for review

## Description of Changes

Adds a strong recommendation to test a newly deployed *Landing Page* using Tor, along with some explanation of why that's important.

The boundary between explanations and instructions is currently unclear in this page (see https://github.com/freedomofpress/securedrop-docs/issues/184). I feel like some explanation were required, but also that concrete instructions would be helpful. Instead of creating two separate sections that would fit awkwardly in the page, I decided to format this new section as a mix of explanation and how-to which can be revisited when #184 is addressed. Thoughts welcome!

Fixes #39 

## Testing
<!-- How should the reviewer test this PR? Write out any special testing steps here. -->

- [ ] Proof-read please :slightly_smiling_face: 
- [ ] Check that the Tor Project link is correct

## Release 
<!-- Any special considerations for release of this change into the stable version of the documentation? -->

_No special considerations for release._


## Checklist (Optional)

- [ ] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [ ] You have previewed (`make docs`) docs at http://localhost:8000

eloquence commented on docs/deployment/landing_page.rst:
```diff
@@ -384,3 +384,27 @@ SecureDrop submissions. We control the servers that store your submissions,
+
+#. Download Tor Browser from the `Tor Project website`_.
+#. Visit your *Landing Page*.
+#. Verify that everything works as expected.
```
I would add a point here about reloading the page using different Tor circuits to observe behavior across a range of exit nodes.

This can be done by clicking the padlock icon on the left side of the address bar, and clicking "New Circuit for this Site".

eloquence replied:
Another thing worth testing is if the page loads correctly if the Tor Browser security settings are set to "Safest".

gonzalo-bulnes replied:
I had a go at both, let me know if that's what you had in mind. : )

eloquence commented on docs/deployment/landing_page.rst:
```diff
@@ -384,3 +384,27 @@ SecureDrop submissions. We control the servers that store your submissions,
+
+The effect of such measures cannot be tested without using Tor, and it is
+a very bad experience for a *source* if visiting a *Landing Page* doesn't work
+as expected. Because of that, it is **strongly recommended** that you test
```
I would change this to active voice (it is recommended -> we recommend).

gonzalo-bulnes replied:
Done.

eloquence commented on docs/deployment/landing_page.rst:
```diff
@@ -384,3 +384,27 @@ SecureDrop submissions. We control the servers that store your submissions,
 
 Do not discuss leaking or whistleblowing, even with trusted contacts.
+
+How to test your *Landing Page* using Tor Browser
```
I agree with you that this whole section could use an information architectural overhaul; as it is, I would prefer to place this new section before the sample landing page content. It is a consideration that should be kept in mind during development and deployment of the landing page, anyway, as it often will require specific configuration changes to the CDN or the webserver to prevent Tor-specific issues.

gonzalo-bulnes replied:
Yep, that makes sense. I moved it just above the content recommendations.

eloquence reviewed (changes requested):
Thanks @gonzalo-bulnes, a few small suggestions but otherwise this looks great!

gonzalo-bulnes commented:
Because it is a short section, I applied your suggestions (and rebased the branch) at once.

I added two links to the relevant Tor Browser documentation for setting the security level and reloading a site using a new Tor circuit. Both pages give a bit of background and explain the different ways to do both. They seem short and useful to me. What do you think @eloquence?

eloquence reviewed (approved):
LGTM, tweaked/consolidated slightly. Thank you!

gonzalo-bulnes commented:
Yes, the single list indeed reads better. 👍
